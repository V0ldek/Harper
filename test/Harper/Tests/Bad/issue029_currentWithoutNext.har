iterate :: Iterator Integer;
iterate = {
    var (i :: Integer) = 0;
    while true {
        yield i;
        i += 1;
    }
};

main :: Integer;
main = iterate.current;