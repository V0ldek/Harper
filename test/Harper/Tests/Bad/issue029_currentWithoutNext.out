Error: iterator's `current` function called before `next` or the sequence had no elements.

Execution terminated with an error: runtime error.